infra: Update some test description which is out of date (#3437)
NVIDIA/TensorRT-LLM: tests/integration/README.md

~~~
.
└── integration              # Root directory for integration tests
    ├── defs            #     Test definitions
    ├── perf_configs    #     Configs for perf tests
    └── test_lists      #     Test lists

>         ├── test-db     #         Test-DB (New test list convention adopted by pytest)

Reviewer: ```suggestion
        ├── test-db     #         Test-DB that is the test list convention adopted by CI
```
